core/rio/api/src/main/java/org/eclipse/rdf4j/rio/RDFFormat.java

```diff
@@ -223,6 +223,18 @@
 			SimpleValueFactory.getInstance().createIRI("http://www.w3.org/ns/formats/JSON-LD"), SUPPORTS_NAMESPACES,
 			SUPPORTS_CONTEXTS, NO_RDF_STAR);
 
+	/**
+	 * The NDJSON-LD</a> is a Newline Delimited JSON-LD format.
+	 * <p>
+	 * The file extension <code>.ndjsonld</code> is recommended for NDJSON-LD documents. The media type is
+	 * <code>application/x-ld+ndjson</code> and the encoding is UTF-8.
+	 * </p>
+	 */
+	public static final RDFFormat NDJSONLD = new RDFFormat("NDJSON-LD", Arrays.asList("application/x-ld+ndjson"),
+			StandardCharsets.UTF_8, Arrays.asList("ndjsonld"),
+			SimpleValueFactory.getInstance().createIRI("http://www.w3.org/ns/formats/NDJSON-LD"), SUPPORTS_NAMESPACES,
```

Comment:
There is no standards URL for this, you can simply pass null (see some of the other formats).